src/NorthwindWebsite/Configuration/ServicesConfiguration.cs

```diff
@@ -101,11 +102,25 @@ public static class ServicesConfiguration
             options.DefaultSignInScheme = IdentityConstants.ExternalScheme;
             options.DefaultSignOutScheme = IdentityConstants.TwoFactorUserIdScheme;
         })
-            .AddCookie(CookieAuthenticationDefaults.AuthenticationScheme)
             .AddCookie(IdentityConstants.ApplicationScheme)
             .AddCookie(IdentityConstants.TwoFactorUserIdScheme)
             .AddExternalCookie();
 
+        services.AddAuthentication().AddMicrosoftAccount(microsoftOptions =>
+        {
+            microsoftOptions.ClientId = appSettings.MicrosoftAccountConfig.ClientId;
+            microsoftOptions.ClientSecret = appSettings.MicrosoftAccountConfig.ClientSecret;
+        });
+
+        services.AddAuthentication().AddAzureAD(options =>
```

Comment:
Probably you may combine `services.AddAuthentication().AddMicrosoftAccount(...).AddAzureAD(...)`